nix/precommit.nix

```diff
@@ -0,0 +1,22 @@
+{ ... }:
+
+{
+  perSystem = { pkgs, ... }: {
+    pre-commit = {
+      inherit pkgs;
+
+      check.enable = true;
+      settings = {
+        hooks = {
+          # Go
+
+
+          # Rust
+          cargo-check.enable = true;
+          clippy.enable = true;
+          rustfmt.enable = true;
```

Comment:
Same here, why don't we move this in the dedicated rust file?

Reply:
Oh damn, I forgot about these. They are already in the rust module, I just forgo to clean them up